### PR TITLE
tests: remove mock SDK client stubs

### PR DESCRIPTION
**Pull request by JadenSimon**
## Problem
We were creating these stub classes to use as mocks but were inconsistent about using them. Mainly because stubbing/mocking frameworks already exist. And they solve the problem better than we do.

## Solution
Just use an existing mocking framework.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/test/ssmDocument/commands/openDocumentItem.test.ts [hunk 1]
```diff
@@ -62,20 +57,23 @@ describe('openDocumentItem', async function () {
...
     })
 
     function generateDocumentItemNode(): DocumentItemNode {
-        const ssmDocumentClient = new MockSsmDocumentClient()
-        sandbox.stub(ssmDocumentClient, 'getDocument').returns(Promise.resolve(rawContent))
+        const ssmDocumentClient = {
+            async getDocument() {
+                return rawContent
+            },
+        } as unknown as SsmDocumentClient
```
Comment (justinmk3) [inline src/test/ssmDocument/commands/openDocumentItem.test.ts:76]: This is a nice use of the escape-hatch that typescript/javascript gives us.

File: src/test/eventSchemas/explorer/registryItemNode.test.ts [hunk 2]
```diff
@@ -16,13 +16,24 @@ import {
...
+function createSchemaClient(data?: { schemas?: Schemas.SchemaSummary[]; registries?: Schemas.RegistrySummary[] }) {
+    return {
+        regionCode: 'code',
+        async *listSchemas(registryName: string, version: string): AsyncIterableIterator<Schemas.SchemaSummary> {
+            yield* data?.schemas ?? []
+        },
+        async *listRegistries() {
+            yield* data?.registries ?? []
+        },
+    } as unknown as SchemaClient
```
Comment (JadenSimon) [inline src/test/eventSchemas/explorer/registryItemNode.test.ts:33]: Usually better to use `sinon` or `mockito` for type-safety though it is kind of nice to just instantiate a stub with data and not worry about the methods being called.

There are ways to generalize this, though by that point you're essentially creating server-side stubs. Which I believe the SDKs are already working on.

Comment (justinmk3) [inline src/test/eventSchemas/explorer/registryItemNode.test.ts:33]: Although we might plan to improve the patterns and practices for testing SDKs / service clients in the future, we can start by mentioning this pattern in our guidelines somewhere (also point to an existing concrete test).

**Review by justinmk3 — APPROVED** [review]
Wonderful!

File: src/test/eventSchemas/explorer/registryItemNode.test.ts (hunk 2 above)
Comment (JadenSimon) [reply src/test/eventSchemas/explorer/registryItemNode.test.ts:33]: If we want to use this pattern, I can write up a factory function real quick that does this with proper types.

File: src/test/ssmDocument/commands/openDocumentItem.test.ts (hunk 1 above)
Comment (JadenSimon) [reply src/test/ssmDocument/commands/openDocumentItem.test.ts:76]: The fact that types and code are decoupled is what makes Typescript so powerful.

File: src/test/eventSchemas/explorer/registryItemNode.test.ts (hunk 2 above)
Comment (JadenSimon) [reply src/test/eventSchemas/explorer/registryItemNode.test.ts:33]: I'll follow up with something more opinionated plus docs in another PR.
Should be able to write a nice, simple abstraction that uses `sinon` but with the class-based ergonomics of `mockito`.

**Pull request merged**